lib/compareLocations.js

```diff
@@ -17,12 +17,14 @@ module.exports = function compareLocations(a, b) {
 		if(typeof b === "string") {
 			return -1;
 		} else if(typeof b === "object") {
-			if(a.start) a = a.start;
-			if(b.start) b = b.start;
-			if(a.line < b.line) return -1;
-			if(a.line > b.line) return 1;
-			if(a.column < b.column) return -1;
-			if(a.column > b.column) return 1;
+			var aa = a.start ? a.start : a;
+			var bb = b.start ? b.start : b;
```

Comment:
I tried `const` but this result in linter warnings because file is not in strict mode. This can be changed to const when file is migrated to ES6.